base_external_api/models/external_api_config.py

```diff
@@ -0,0 +1,157 @@
+        ctx = self.env.context
+        active_id = ctx.get("active_id") or ctx.get("params", {}).get("id")
+        active_model = ctx.get("active_model") or ctx.get("params", {}).get("model")
+        return self.env["external.api.log"].create(
+            {
+                "api_id": self.id,
+                "datetime": datetime.datetime.now(),
+                "user_id": self.env.user.id,
+                "executed_request": f"requests.{method}({url}, {kwargs})",
+                "execution_record": f"{active_model}({active_id})"
+                if active_model and active_id
+                else False,
+            }
+        )
+
+    @api.model
+    def _update_log(self, log, response):
+        log.write(
+            {
+                "status": "success" if response.ok else "error",
+                "status_code": response.status_code,
+                "response": response.text if not response.ok else "",
+            }
+        )
+
+    def _call_and_update_log(self, method, url, log, **kwargs):
+        request_func = getattr(requests, method)
+        res = False
+        try:
+            res = request_func(url=url, **kwargs)
+        except requests.exceptions.Timeout as error:
+            new_cr = Registry(self.env.cr.dbname).cursor()
+            env = api.Environment(new_cr, self.env.uid, self.env.context)
+            log = env["external.api.log"].browse(log.id)
+            log.write({"status": "exception", "response": error})
+            new_cr.commit()
+            new_cr.close()
+            raise RetryableJobError(
+                "Timeout connecting remote server. Must be retried later"
+            ) from error
+        except requests.exceptions.RequestException as error:
+            log.write({"status": "exception", "response": error})
+        else:
+            log.write(
+                {
+                    "status": "success" if res.ok else "http_error",
+                    "status_code": res.status_code,
+                    "response": res.text if not res.ok else "",
+                }
+            )
+        return res
+
+    def call(self, method, url, queued=False, **kwargs):
+        self.ensure_one()
+        if self.state != "production":
+            res = False
+        else:
+            url = self._parse_url(self.base_url) + self._parse_url(url)
+            updated_kwargs = self._update_kwargs(**kwargs)
+            log = self._create_log(method, url, **updated_kwargs)
+            res = self._call_and_update_log(method, url, log, **updated_kwargs)
+        return res
+
+    def queued_call(self, method, url, **kwargs):
+        self.ensure_one()
+        if self.state != "production":
+            job = False
+        else:
+            url = self._parse_url(self.base_url) + self._parse_url(url)
+            updated_kwargs = self._update_kwargs(**kwargs)
+            log = self._create_log(method, url, **updated_kwargs)
+            job = self.with_delay()._call_and_update_log(
```

Comment:
is posible add delay in the configuration?

Reply:
Since you are taking it into account in the calls, I would allow it to be configured.